### PR TITLE
 tests sharing between database and LDAP users 

### PR DESCRIPTION
**Pull request by individual-it**
check if database and LDAP users can interact correctly

### CODE REVIEW

**Comment by dpakach** [comment]
LGTM :+1:

File: tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature [hunk 1]
```diff
@@ -0,0 +1,125 @@
+@api
+Feature: Sharing between local and LDAP users
+  As a user
+  I want to be able to share files and folders with any user regardless of the backend
+
+  Background:
+    Given user "local-user" has been created with default attributes in the database user backend
+    And these users have been initialized:
+    #this are LDAP users and are not initialized yet
```
Comment (phil-davis) [inline tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature]: s/this/these

File: tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      | username | password |
+      | user0    | 123456   |
```
Comment (phil-davis) [inline tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature]: why do we need to put the password in this table? 
doesn't the code underneath already know the "default" password that goes with each user?

File: tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    """
+    ownCloud test text file parent
+    
+    """
+    And the content of file "/PARENT (2)/parent.txt" for user "user2" should be:
+    """
+    ownCloud test text file parent
+    
+    """
+    When the administrator removes user "user1" from ldap group "grp1"
```
Comment (phil-davis) [inline tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature]: Can you do this?
I thought that LDAP group membership comes read-only from the LDAP server.

Comment (individual-it) [reply tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature]: yes, but this step does manipulate the LDAP server itself, it does not do anything with the oC server.

File: tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature (hunk 2 above)
Comment (individual-it) [reply tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature]: this particular does not have the magic, I will have to make a PR to core to change it

File: tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature (hunk 3 above)
Comment (phil-davis) [reply tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature]: :+1:

File: tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature (hunk 2 above)
Comment (individual-it) [reply tests/acceptance/features/apiUserLDAPSharing/sharingLocalUserLdapUser.feature]: PR here https://github.com/owncloud/core/pull/34213 , @phil-davis please review

**Comment by phil-davis** [comment]
@individual-it educate me - where in ``.drone.yml`` do the suites ``apiUserLDAPConnection`` and ``apiUserLDAPSharing`` get run?

**Comment by individual-it** [comment]
@phil-davis in the ` #acceptance tests - ldap specific` section

**Comment by phil-davis** [comment]
Looks better now. I must have an out-of-date local git clone, because I didn't see ``apiUserLDAPConnection`` in my ``.drone.yml``

Will need a rebase tomorrow after the QA tarball comes.

**Pull request merged**